pkg/kubelet/config/file_test.go

```diff
@@ -54,30 +58,111 @@ func TestUpdateOnNonExistentFile(t *testing.T) {
 		}
 
 	case <-time.After(wait.ForeverTestTimeout):
-		t.Errorf("Expected update, timeout instead")
+		t.Fatalf("Expected update, timeout instead")
 	}
 }
 
-func writeTestFile(t *testing.T, dir, name string, contents string) *os.File {
-	file, err := ioutil.TempFile(os.TempDir(), "test_pod_config")
+func TestReadPodsFromFileExistAlready(t *testing.T) {
+	hostname := "random-test-hostname"
+	var testCases = getTestCases(hostname)
+
+	for _, testCase := range testCases {
+		func() {
+			dirName, err := utiltesting.MkTmpdir("file-test")
+			if err != nil {
+				t.Fatalf("Unable to create temp dir: %v", err)
+			}
+			defer os.RemoveAll(dirName)
+			file := testCase.writeToFile(dirName, "test_pod_config", t)
+
+			ch := make(chan interface{})
+			NewSourceFile(file.Name(), hostname, time.Millisecond, ch)
+			select {
+			case got := <-ch:
+				update := got.(kubetypes.PodUpdate)
+				for _, pod := range update.Pods {
+					if errs := validation.ValidatePod(pod); len(errs) > 0 {
+						t.Errorf("%s: Invalid pod %#v, %#v", testCase.desc, pod, errs)
+					}
+				}
+				if !api.Semantic.DeepEqual(testCase.expected, update) {
+					t.Errorf("%s: Expected %#v, Got %#v", testCase.desc, testCase.expected, update)
+				}
+			case <-time.After(wait.ForeverTestTimeout):
+				t.Fatalf("%s: Expected update, timeout instead", testCase.desc)
+			}
+		}()
+	}
+}
+
+func TestReadPodsFromFileExistLater(t *testing.T) {
```

Comment:
`TestReadPodsFromFileExistLater` is a case in which the watched file doesn't exist at the first beginning. As a result, the inotify will not be initialized at that time. Instead, a [loop](https://github.com/wu8685/kubernetes/blob/5479a87835891edf8805cedb18c82bf48b4d86e1/pkg/kubelet/config/file.go#L47) will periodically try to watch the path. (this loop is a key part making the past version work) The period is `1 second` in this test case. It is why this case alway take 1s to finish.

I changed the period to `100*time.Millisecond` [here](https://github.com/wu8685/kubernetes/blob/5479a87835891edf8805cedb18c82bf48b4d86e1/pkg/kubelet/config/file_linux_test.go#L252)